### PR TITLE
Mec172x rtos timer

### PR DESCRIPTION
**Pull request by scottwcpg**
Depends on PR 37125
Update the Microchip XEC RTOS timer driver used by both MEC152x and MEC172x. The updates make use of device tree parameters that have been added to MEC172x and MEC152x DT and YAML.  Other commits in this PR add the MEC172x SoC Kconfig logic and board level Kconfig logic.

### CODE REVIEW

**Comment by cfriedt** [comment]
@scottwcpg - can you please squash your relevant commits into one?

See https://docs.zephyrproject.org/latest/contribute/index.html#other-commit-expectations

Based on this and your previous PR's it appears that you are branching from branches when making PR's rather than branching from main. Please ensure you branch from main.

See https://docs.zephyrproject.org/latest/contribute/index.html#contribution-workflow

That will mitigate the "Depends on xxxxx" that we're seeing. These two PR's are actually independent logical additions, so they should each apply independently and cleanly to main.

**Review by cfriedt — CHANGES_REQUESTED** [review]
Commit guidelines & workflow must be followed.

Typically we see at most 2 commits per change: 1 commit for code, and 1 commit for tests.

**Comment by scottwcpg** [comment]
@cfriedt Squashed all commits into one as requested.

File: drivers/timer/mchp_xec_rtos_timer.c [hunk 1]
```diff
@@ -50,14 +52,30 @@ BUILD_ASSERT(CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC == 32768,
 	(CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC / CONFIG_SYS_CLOCK_TICKS_PER_SEC)
 
 #define TIMER_REGS	\
-	((RTMR_Type *) DT_INST_REG_ADDR(0))
+	((struct rtmr_regs *) DT_INST_REG_ADDR(0))
 
-/* Mask off bits[31:28] of 32-bit count */
-#define TIMER_MAX	0x0FFFFFFFUL
+#define ECIA_XEC_REGS	\
+	((struct ecia_regs *) DT_REG_ADDR(DT_NODELABEL(ecia)))
```
Comment (jackrosenthal) [inline drivers/timer/mchp_xec_rtos_timer.c:58]: no space after cast

File: drivers/timer/mchp_xec_rtos_timer.c [hunk 2]
```diff
@@ -50,14 +52,30 @@ BUILD_ASSERT(CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC == 32768,
...
 
-#define TIMER_COUNT_MASK	0x0FFFFFFFUL
+#ifdef CONFIG_ARCH_HAS_CUSTOM_BUSY_WAIT
+#define PCR_XEC_REGS	\
+	((struct pcr_regs *) DT_REG_ADDR(DT_NODELABEL(pcr)))
```
Comment (jackrosenthal) [inline drivers/timer/mchp_xec_rtos_timer.c:62]: ditto

File: drivers/timer/mchp_xec_rtos_timer.c [hunk 3]
```diff
@@ -271,7 +323,7 @@ static void xec_rtos_timer_isr(const void *arg)
 
 uint32_t sys_clock_elapsed(void)
 {
-	return 0U;
+	return 0u;
```
Comment (jackrosenthal) [inline drivers/timer/mchp_xec_rtos_timer.c:326]: please make unrelated style changes in a separate PR

**Review by jackrosenthal — CHANGES_REQUESTED** [review]

File: drivers/timer/mchp_xec_rtos_timer.c (hunk 2 above)
Comment (scottwcpg) [reply drivers/timer/mchp_xec_rtos_timer.c]: Done

File: drivers/timer/mchp_xec_rtos_timer.c (hunk 1 above)
Comment (scottwcpg) [reply drivers/timer/mchp_xec_rtos_timer.c]: Done

File: drivers/timer/mchp_xec_rtos_timer.c (hunk 3 above)
Comment (scottwcpg) [reply drivers/timer/mchp_xec_rtos_timer.c]: Put it back to 0U until future style PR.

**Review by jackrosenthal — APPROVED** [review]

**Review by cfriedt — APPROVED** [review]
